### ISSUE TITLE
Failed to send configuration to localhost

### ISSUE THREAD

**Issue by maxiride** [lucaslorentz/caddy-docker-proxy#205]
I'm once again trying to automatically handle the reverse proxying of some services.

The caddy-docker-proxy image correctly recognize the labels and create the correct Caddyfile, however it throws the following error and the services aren't reachable:

```
2020/12/14 19:48:09 [INFO] Sending configuration to localhost,
2020/12/14 19:48:39 [ERROR] Failed to send configuration to localhost: Post "http://localhost:2019/load": dial tcp 40.68.249.35:2019: i/o timeout
```
I suppose the issue lies in the resolved localhost ip: 40.68.249.35 which is wrong. The container's IP in the caddy network is 172.24.0.4

I've spun up the following compose file:
```yml
version: "3.7"

services:
  caddy2:
    image: lucaslorentz/caddy-docker-proxy:2.3-alpine
    container_name: caddy2
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - caddy_data:/data
      - caddy_config:/config
    networks:
      - caddy

  whoami:
    image: jwilder/whoami
    networks:
      - caddy
    labels:
      caddy: :80
      caddy.reverse_proxy: "{{upstreams 8000}}"
      caddy.tls: "internal"

volumes:
  caddy_data:
  caddy_config:

networks:
  caddy:
    driver: bridge
```

And this is the complete log:
```
2020/12/15 11:00:14 [INFO] Running caddy proxy server,
2020/12/15 11:00:14 [INFO] Running caddy proxy controller,
2020/12/15 11:00:14 [INFO] CaddyfilePath: ,
2020/12/15 11:00:14 [INFO] LabelPrefix: caddy,
2020/12/15 11:00:14 [INFO] PollingInterval: 30s,
2020/12/15 11:00:14 [INFO] ProcessCaddyfile: true,
2020/12/15 11:00:14 [INFO] ProxyServiceTasks: true,
2020/12/15 11:00:14 [INFO] IngressNetworks: [],
2020/12/15 11:00:14 [INFO] Caddy ContainerID: 6f5d0c3805f89aef3c75baaf87b6289399d7199bb231cb609dfe7a73edadee32,
2020/12/15 11:00:14 [INFO] IngressNetworksMap: map[dbcee0f91a1cd24a299cd87e0ba056b152256b116a4f8f10fef6eed9585ec4e2:true],
2020/12/15 11:00:14 [INFO] Swarm is available: false,
2020/12/15 11:00:14 [INFO] Skipping default Caddyfile because no path is set,
[INFO] Skipping configs because swarm is not available,
[INFO] Skipping services because swarm is not available,
2020/12/15 11:00:14 [INFO] New Caddyfile:,
:80 {,
	reverse_proxy 172.24.0.2:8000,
	tls internal,
},
2020/12/15 11:00:14 [INFO] New Config JSON:,
{"apps":{"http":{"servers":{"srv0":{"listen":[":80"],"routes":[{"handle":[{"handler":"reverse_proxy","upstreams":[{"dial":"172.24.0.2:8000"}]}]}]}}},"tls":{"automation":{"policies":[{"issuer":{"module":"internal"}}]}}}},
2020/12/15 11:00:14 [INFO] Sending configuration to localhost,
2020/12/15 11:00:44 [ERROR] Failed to send configuration to localhost: Post "http://localhost:2019/load": dial tcp 40.68.249.35:2019: i/o timeout,
2020/12/15 11:01:14 [INFO] Sending configuration to localhost,
```

**Comment by regbo** [comment]
@maxiride I posted a PR with a fix. for now you can find a working dockerfile here:
https://github.com/lucaslorentz/caddy-docker-proxy/issues/238#issuecomment-825195742

**Comment by maxiride** [comment]
> @maxiride I posted a PR with a fix. for now you can find a working dockerfile here:
> [#238 (comment)](https://github.com/lucaslorentz/caddy-docker-proxy/issues/238#issuecomment-825195742)

Thanks for the update!
I have to say that after days of desperation I reinstalled Docker and it solved my issue. Reinstalling docker obviously has nothing to do with this docker image but it kinda reset everything else too.

**Comment by lucaslorentz** [comment]
@maxiride 
Interesting that it solved automatically.
Usually localhost is resolved with the loopback 127.0.0.1, I don't understand under which circumstances docker would override it to be something else, specially some IP that doesn't exists.

**Comment by maxiride** [comment]
> 
> 
> @maxiride
> Interesting that it solved automatically.
> Usually, localhost is resolved with the loopback 127.0.0.1, I don't understand under which circumstances docker would override it to be something else, especially some IP that doesn't exist.

I'm very confused too, I searched everywhere on the internet without even finding a small hint on the subject. It surely is a very isolated case and most likely is affected by other factors outside this image scope.

For what's worth I suspect the issue could be closed as its a bit of a dead end. I can't reproduce the issue nor provide further details.

**Comment by lucaslorentz** [comment]
Thanks @maxiride 
Closing it.
